server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -1568,4 +1570,22 @@ public static ZonedDateTime toZonedDateTime(TemporalAccessor accessor, ZonedDate
 
         return result;
     }
+
+    public static DateTimeZone zoneIdToTimeZone(ZoneId zoneId) {
```

Comment:
can you rename to `zoneIdToDateTimeZone()` to prevent people thinking that this has something to do with the old java TimeZone classes?

Reply:
I think these two methods will fail when you create a test that feeds a random `ZoneId` here and a random `DateTimeZone` to the method below as timezones are not the same. `EST` for example IIRC exists only in joda time.

Reply:
I was thinking about adding some tests to double check this. AFAICT from looking at the implementation, both use the same timezone data files that java's TimeZone uses. The only difference I could find is with ZoneOffset in java time (and the equivalent in joda), the time offset is parsed locally, not using any standard lookup. And in the case of java time, it potentially outputs a "non iso" case where an offset contains seconds. Also, a zero offset in java time prints to just "Z", while in joda time this prints as "UTC". This is the reason for the special casing offsets. But the named ids should be the same.

Reply:
```
    public void testJavaToJodaTimeZones() {
        Set<String> availableZoneIds = ZoneId.getAvailableZoneIds();
        List<String> brokenIds = availableZoneIds.stream().filter(id -> {
            try {
                DateTimeZone.forID(id);
                return false;
            } catch (Exception e) {
                return true;
            }
        }).collect(Collectors.toList());
        assertThat(brokenIds, equalTo(Collections.emptyList()));
    }

    public void testJodaToJavaTimeZones() {
        Set<String> jodaIds = DateTimeZone.getAvailableIDs();
        List<String> brokenIds = jodaIds.stream().filter(id -> {
            try {
                ZoneId.of(id);
                return false;
            } catch (Exception e) {
                return true;
            }
        }).collect(Collectors.toList());
        assertThat(brokenIds, equalTo(Collections.emptyList()));
    }
```

both of these fail. JavaTimeToJoda seems not so much of an issue, as all the dates are to me unused ones starting with `SystemV`.

The second test though returns `EST, HST, MST, ROC`, where I could imagine that EST is something used in the US?

Reply:
Thanks. I will investigate this more and figure out how joda is supporting these additional ids that are supposedly not in the common timezone database.

Reply:
I added a test for the joda to java conversion, and deprecation warnings for those that will not work when we switch to java time.